pkg/jobs/structured_log.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package jobs
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/redact"
+)
+
+type JobEventType string
+
+const (
+	StateChange JobEventType = "state_change"
+)
+
+type JobEvent struct {
+	JobID        jobspb.JobID `json:"job_id"`
+	EventType    JobEventType `json:"event_type"`
```

Comment:
should we add some notion of notability or severity to these events to allow a reader of this log to more easily read for only notable events (like state changes or resuming after retries) vs all events?

I know we only have one event type now, but we might want to be writing it with some sort of level attached now so that when we start writing more events we have a level against which to set those?